rmf_site_format/src/workcell.rs

```diff
@@ -280,6 +280,137 @@ pub struct Workcell {
+                size: urdf_rs::Vec3([x as f64, y as f64, z as f64]),
+            },
+            Geometry::Primitive(MeshPrimitive::Cylinder { radius, length }) => {
+                urdf_rs::Geometry::Cylinder {
+                    radius: radius as f64,
+                    length: length as f64,
+                }
+            }
+            Geometry::Primitive(MeshPrimitive::Capsule { radius, length }) => {
+                urdf_rs::Geometry::Capsule {
+                    radius: radius as f64,
+                    length: length as f64,
+                }
+            }
+            Geometry::Primitive(MeshPrimitive::Sphere { radius }) => urdf_rs::Geometry::Sphere {
+                radius: radius as f64,
+            },
+            _ => todo!("Only meshes and primitives are supported for conversion to urdf geometry"),
+        }
+    }
+}
+
+impl From<Workcell> for urdf_rs::Robot {
+    fn from(workcell: Workcell) -> Self {
+        let visuals_and_parents: Vec<(urdf_rs::Visual, _)> = workcell
+            .visuals
+            .into_iter()
+            .map(|v| {
+                let visual = v.1.bundle;
+                let visual = urdf_rs::Visual {
+                    name: Some(visual.name),
+                    origin: visual.pose.into(),
+                    geometry: visual.geometry.into(),
+                    material: None,
+                };
+                (visual, v.1.parent)
+            })
+            .collect();
+        let collisions_and_parents: Vec<(urdf_rs::Collision, _)> = workcell
+            .collisions
+            .into_iter()
+            .map(|c| {
+                let collision = c.1.bundle;
+                let collision = urdf_rs::Collision {
+                    name: Some(collision.name),
+                    origin: collision.pose.into(),
+                    geometry: collision.geometry.into(),
+                };
+                (collision, c.1.parent)
+            })
+            .collect();
+        urdf_rs::Robot {
+            name: workcell.properties.name,
+            links: workcell
+                .frames
+                .into_iter()
+                .map(|f| {
+                    let frame = f.1.bundle;
+
+                    let visual: Vec<Visual> = visuals_and_parents
+                        .iter()
+                        .filter(|(_, parent)| parent == &f.1.parent)
+                        .map(|(visual, _)| visual.clone())
+                        .collect();
+                    let collision: Vec<Collision> = collisions_and_parents
+                        .iter()
+                        .filter(|(_, parent)| parent == &f.1.parent)
+                        .map(|(collision, _)| collision.clone())
+                        .collect();
+
+                    let pose: urdf_rs::Pose = match frame.anchor {
+                        Anchor::Pose3D(pose) => pose.into(),
+                        _ => todo!(),
```

Comment:
Let's make sure we implement this before we merge.

Reply:
Done in dfbf3d3.